### PR TITLE
Extract JOIN in own plan step

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Extract JOIN action from before_join. So we could make own query plan step just for JOIN.

### CODE REVIEW

File: src/Interpreters/ExpressionAnalyzer.cpp [hunk 1]
```diff
@@ -502,18 +493,20 @@ void ExpressionAnalyzer::addJoinAction(ExpressionActionsPtr & actions, JoinPtr j
...
-    if (!ast_join)
-        return false;
+    ExpressionActionsChain::Step & step = chain.lastStep(sourceColumns());
+
+    getRootActions(analyzedJoin().leftKeysList(), only_types, step.actions);
+    return true;
+}
 
-    JoinPtr table_join = makeTableJoin(*ast_join);
+bool SelectQueryExpressionAnalyzer::appendJoin(ExpressionActionsChain & chain)
```
Comment (KochetovNicolai) [inline src/Interpreters/ExpressionAnalyzer.cpp:504]: Result is always `true`. This two methods could return `void` as well.
It is also good to check if we have `JOIN` indeed.

Comment (4ertus2) [reply src/Interpreters/ExpressionAnalyzer.cpp:504]: There're some methods that do the same. For now `ExpressionAnalyzer::append*()` is expected to return bool. The check if outside of this method, It's private and called from one place, just after the check if we have a join. I'd prefer not to change it in this PR. It's TODO for future refactoring.

**Review by KochetovNicolai — APPROVED** [review]

File: src/Interpreters/ExpressionAnalyzer.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Interpreters/ExpressionAnalyzer.cpp:504]: ok

**Comment by KochetovNicolai** [comment]
Not waiting for tests because only comment was changed.

**Pull request merged**